spec/unit/puppet/type/registry_value_spec.rb

```diff
@@ -60,12 +61,50 @@
     it 'should validate the length of the value data'
     it 'should be case-preserving'
     it 'should be case-insensitive'
-    it 'should autorequire ancestor keys'
     it 'should support 32-bit values' do
       value = described_class.new(:path => '32:hklm\software\foo', :catalog => catalog)
     end
   end
 
+  describe '#autorequire' do
+    let(:subject) { described_class.new(:title => subject_title, :catalog => catalog) }
+    [
+      {
+        :context                => 'with a non-default value_name',
+        :reg_value_title        => 'hklm\software\foo\bar',
+        :expected_reg_key_title => 'hklm\Software\foo',
+      },
+      {
+        :context                => 'with a mixed case path and value_name',
+        :reg_value_title        => 'hkLm\soFtwarE\fOo\Bar',
+        :expected_reg_key_title => 'hklm\Software\foo',
+      },
+      {
+        :context                => 'with a default value_name',
+        :reg_value_title        => 'hklm\software\foo\bar\\',
+        :expected_reg_key_title => 'hklm\Software\foo\bar',
+      }
+    ].each do |testcase|
+      context testcase[:context] do
+        let(:subject) { described_class.new(:title => testcase[:reg_value_title], :catalog => catalog) }
+
+        it 'should not autorequire ancestor keys if none exist' do
+          expect(subject.autorequire).to eq([])
+        end
+
+        it 'should only autorequire the immediate ancestor registry_key resource' do
+          catalog.add_resource(Puppet::Type.type(:registry_key).new(:path => 'hklm\Software', :catalog => catalog))
+          catalog.add_resource(Puppet::Type.type(:registry_key).new(:path => 'hklm\Software\foo', :catalog => catalog))
+          catalog.add_resource(Puppet::Type.type(:registry_key).new(:path => 'hklm\Software\foo\bar', :catalog => catalog))
```

Comment:
Add one new test here like:

```
catalog.add_resource(Puppet::Type.type(:registry_key).new(:path => 'hklm\Software\foo\bar\baz\ethan', :catalog => catalog))
```
And make sure that it autorequires nearest value of 'hklm\Software\foo\bar'